docs/subnets/hello-world-precompile-tutorial.md

```diff
@@ -548,201 +851,153 @@ func setGreeting(accessibleState PrecompileAccessibleState, caller common.Addres
-We use this to check out whether the precompile is enabled. We defined `GetHelloWorldConfig()` in
-the last step. The below code snippet can be copied and pasted to overwrite the default code:
-
-```go
-// IsHelloWorld returns whether [blockTimestamp] is either equal to the HelloWorld
-// fork block timestamp or greater.
-func (c *ChainConfig) IsHelloWorld(blockTimestamp *big.Int) bool {
-	config := c.GetHelloWorldConfig(blockTimestamp)
-	return config != nil && !config.Disable
+// PrecompileTest is a test case for a precompile
+type PrecompileTest struct {
+	// Caller is the address of the precompile caller
+	Caller common.Address
+	// Input the raw input bytes to the precompile
+	Input []byte
+	// InputFn is a function that returns the raw input bytes to the precompile
+	// If specified, Input will be ignored.
+	InputFn func(t *testing.T) []byte
+	// SuppliedGas is the amount of gas supplied to the precompile
+	SuppliedGas uint64
+	// ReadOnly is whether the precompile should be called in read only
+	// mode. If true, the precompile should not modify the state.
+	ReadOnly bool
+	// Config is the config to use for the precompile
+	// It should be the same precompile config that is used in the
+	// precompile's configurator.
+	// If nil, Configure will not be called.
+	Config precompileconfig.Config
+	// BeforeHook is called before the precompile is called.
+	BeforeHook func(t *testing.T, state contract.StateDB)
+	// AfterHook is called after the precompile is called.
+	AfterHook func(t *testing.T, state contract.StateDB)
+	// ExpectedRes is the expected raw byte result returned by the precompile
+	ExpectedRes []byte
+	// ExpectedErr is the expected error returned by the precompile
+	ExpectedErr string
+	// BlockNumber is the block number to use for the precompile's block context
+	BlockNumber int64
 }
 ```
 
-We can now add it to the `AvalancheRules()` which creates and returns a new instance of `Rules`. The
-below code snippet can be copied and pasted to overwrite the default code:
+Each test can populate the fields of the `PrecompileTest` struct to customize the test.
+This test uses an AllowList helper function
+`allowlist.RunPrecompileWithAllowListTests(t, Module, state.NewTestStateDB, tests)`
+which can run all specified tests plus AllowList test suites. If you don't plan to use AllowList,
+you can directly run them as follows:
 
 ```go
-rules.IsContractNativeMinterEnabled = c.IsContractNativeMinter(blockTimestamp)
-rules.IsTxAllowListEnabled = c.IsTxAllowList(blockTimestamp)
-rules.IsFeeConfigManagerEnabled = c.IsFeeConfigManager(blockTimestamp)
-rules.IsHelloWorldEnabled = c.IsHelloWorld(blockTimestamp)
-// ADD YOUR PRECOMPILE HERE
-// rules.Is{YourPrecompile}Enabled = c.{IsYourPrecompile}(blockTimestamp)
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			test.Run(t, module, newStateDB(t))
+		})
+	}
 ```
 
-Done!
+You can check other `contract_test.go` files in the `/precompile/contracts`.
 
```

Comment:
nit: link here?

Reply:
moved that to above. Since there are multiple files like that, I don't think we can show a link. But I put a link for hello world tests right after this one. so should be clear enough.